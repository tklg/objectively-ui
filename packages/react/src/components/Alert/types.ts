import { ForwardRefExoticComponent, ReactElement } from 'react'
import { PropsWithChildrenAndClassName } from 'src/types/PropsWithChildrenAndClassName'
import { StatusColor } from 'src/types/props'

export interface AlertProps extends PropsWithChildrenAndClassName {
  severity?: StatusColor;
  title?: string;
  action?: ReactElement;
}
export type AlertComponent<P = AlertProps> = ForwardRefExoticComponent<P>
