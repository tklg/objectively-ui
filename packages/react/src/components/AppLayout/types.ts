import { ReactNode } from 'react'

export interface AppLayoutProps {
  className?: string;
  leftNavigation?: ReactNode;
  rightNavigation?: ReactNode;
  content?: ReactNode;
  header?: ReactNode;
  footer?: ReactNode;
  fullWidth?: boolean;
  maxWidth?: string | number;
}

export interface AppLayoutContextProps {
  maxWidth: string;
}
