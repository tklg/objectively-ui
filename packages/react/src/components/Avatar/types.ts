import { CommonColor, CommonSize } from 'src/types/props'

export interface AvatarProps {
  size?: CommonSize;
  src?: string;
  name?: string;
  initials?: string;
  className?: string;
  color?: CommonColor | 'auto';
}
