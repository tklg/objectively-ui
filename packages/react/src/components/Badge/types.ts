import { ForwardRefExoticComponent, ReactElement } from 'react'
import { CommonColor } from 'src/types/props'

export interface BadgeProps {
  children: ReactElement;
  className?: string;
  fit?: 'square' | 'circle';
  value?: string | number;
  placement?: 'top-right' | 'bottom-right' | 'bottom-left' | 'top-left';
  color?: CommonColor;
}

export type BadgeComponent<P = BadgeProps> = ForwardRefExoticComponent<P>
