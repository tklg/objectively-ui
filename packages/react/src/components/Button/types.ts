import { ButtonHTMLAttributes, DetailedHTMLProps, ReactNode } from 'react'
import { CommonSize } from 'src/types/sizes'

export type ButtonVariant = 'default' | 'primary' | 'text' | 'link';

export interface ButtonProps extends DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
  size?: CommonSize;
  fullWidth?: boolean;
  iconStart?: ReactNode;
  variant?: ButtonVariant;
  danger?: boolean;
}
