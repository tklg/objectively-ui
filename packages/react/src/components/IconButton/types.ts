import { ForwardRefExoticComponent, ReactNode } from 'react'
import { ButtonBaseProps } from 'src/components/Button/ButtonBase.types'

export type IconButtonVariant = 'default' | 'solid' | 'text'

export interface IconButtonProps extends Omit<ButtonBaseProps, 'children'> {
  icon?: ReactNode;
  variant?: IconButtonVariant;
}

export type IconButtonComponent<P = IconButtonProps> = ForwardRefExoticComponent<P>
