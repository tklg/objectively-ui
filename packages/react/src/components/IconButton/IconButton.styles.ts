import { css } from '@emotion/react'
import { ColorTheme } from 'src/types/ColorTheme'
import { CommonSize } from 'src/types/props'

export const iconButtonStyles = (theme: ColorTheme, size: CommonSize) => css({
  width: theme.size[size],
  padding: 0,
})
