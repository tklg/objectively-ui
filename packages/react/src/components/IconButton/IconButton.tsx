import { forwardRef } from 'react'
import { ButtonBase } from 'src/components/Button/ButtonBase'
import { iconButtonStyles } from 'src/components/IconButton/IconButton.styles'
import { IconButtonComponent, IconButtonProps } from 'src/components/IconButton/types'
import { useTheme } from 'src/hooks'
import { buildClassName } from 'src/utils/buildClassName'

const ELEMENT_NAME = 'IconButton'

export const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(({
  size = 'md',
  variant = 'default',
  color = 'default',
  icon,
  className: _className,
  ...props
}, ref) => {
  const theme = useTheme()

  const className = buildClassName(ELEMENT_NAME, {
    size,
    variant,
    color,
  }, _className)

  return (
    <ButtonBase
      ref={ref}
      {...props}
      className={className}
      css={iconButtonStyles(theme, size)}
      size={size}
      color={color}
      variant={variant}
    >
      {icon}
    </ButtonBase>
  )
}) as IconButtonComponent

IconButton.displayName = ELEMENT_NAME

if (process.env.NODE_ENV !== 'production') {
  IconButton.displayName = ELEMENT_NAME
}
