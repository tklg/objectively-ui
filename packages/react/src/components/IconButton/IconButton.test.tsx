import { IconButton } from 'src/components/IconButton/IconButton'
import { IconButtonVariant } from 'src/components/IconButton/types'
import { CommonColor, CommonSize } from 'src/types/props'
import { render, screen } from 'src/__test__/render'

describe('IconButton', () => {
  it('renders an iconbutton', () => {
    render(<IconButton icon='icon' />)
    expect(screen.getByRole('button')).toHaveTextContent('icon')
  })

  it('renders a disabled iconbutton', () => {
    render(<IconButton icon='icon' disabled />)
    expect(screen.getByRole('button')).toBeDisabled()
  })

  it('with an icon', () => {
    render(<IconButton icon='icon' />)
    expect(screen.getByRole('button')).toHaveTextContent('icon')
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  describe('sizes', () => {
    it.each([['sm'], ['md'], ['lg']])('%s', (args) => {
      const size = args[0] as CommonSize

      render(<IconButton icon='icon' size={size} />)
      expect(screen.getByRole('button')).toMatchSnapshot(size)
    })
  })

  describe('variants', () => {
    it.each([['default'], ['solid'], ['text']])('%s', (args) => {
      const variant = args[0] as IconButtonVariant

      render(<IconButton icon='icon' variant={variant} />)
      expect(screen.getByRole('button')).toMatchSnapshot(variant)
    })
  })

  describe('colors', () => {
    it.each([['default'], ['primary'], ['secondary'], ['info'], ['warning'], ['error'], ['success']])('%s', (args) => {
      const color = args[0] as CommonColor

      render(<IconButton icon='icon' color={color} />)
      expect(screen.getByRole('button')).toMatchSnapshot(color)
    })
  })
})
