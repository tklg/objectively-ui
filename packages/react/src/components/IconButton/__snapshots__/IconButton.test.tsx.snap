// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton colors default: d 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorD ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantDefault ObjUI-IconButton-colorD emotion-0"
>
  icon
</button>
`;

exports[`IconButton colors error: e 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorE ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantDefault ObjUI-IconButton-colorE emotion-0"
>
  icon
</button>
`;

exports[`IconButton colors info: i 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorI ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantDefault ObjUI-IconButton-colorI emotion-0"
>
  icon
</button>
`;

exports[`IconButton colors primary: p 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorP ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantDefault ObjUI-IconButton-colorP emotion-0"
>
  icon
</button>
`;

exports[`IconButton colors secondary: s 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorS ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantDefault ObjUI-IconButton-colorS emotion-0"
>
  icon
</button>
`;

exports[`IconButton colors success: s 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorS ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantDefault ObjUI-IconButton-colorS emotion-0"
>
  icon
</button>
`;

exports[`IconButton colors warning: w 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorW ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantDefault ObjUI-IconButton-colorW emotion-0"
>
  icon
</button>
`;

exports[`IconButton sizes lg: l 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeL ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeL ObjUI-IconButton-variantDefault ObjUI-IconButton-colorDefault emotion-0"
>
  icon
</button>
`;

exports[`IconButton sizes md: m 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeM ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeM ObjUI-IconButton-variantDefault ObjUI-IconButton-colorDefault emotion-0"
>
  icon
</button>
`;

exports[`IconButton sizes sm: s 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeS ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeS ObjUI-IconButton-variantDefault ObjUI-IconButton-colorDefault emotion-0"
>
  icon
</button>
`;

exports[`IconButton variants default: d 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantD ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantD ObjUI-IconButton-colorDefault emotion-0"
>
  icon
</button>
`;

exports[`IconButton variants solid: s 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantS ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantS ObjUI-IconButton-colorDefault emotion-0"
>
  icon
</button>
`;

exports[`IconButton variants text: t 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantT ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantT ObjUI-IconButton-colorDefault emotion-0"
>
  icon
</button>
`;

exports[`IconButton with an icon 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-IconButton ObjUI-IconButton-sizeMd ObjUI-IconButton-variantDefault ObjUI-IconButton-colorDefault emotion-0"
>
  icon
</button>
`;
